File: java/src/test/java/software/amazon/samples/awscdkassertionssamples/DeadLetterQueueTest.java
package software.amazon.samples.awscdkassertionssamples;

import org.junit.jupiter.api.Test;
import software.amazon.awscdk.assertions.Match;
import software.amazon.awscdk.assertions.Template;
import software.amazon.awscdk.core.Stack;

import java.util.Collections;
import java.util.Map;

public class DeadLetterQueueTest {
    @Test
    public void testCreatesAlarm() {
        final Stack stack = new Stack();
        new DeadLetterQueue(stack, "DeadLetterQueue");

        final Template template = Template.fromStack(stack);
        template.hasResourceProperties("AWS::CloudWatch::Alarm", Map.of(
                "Namespace", "AWS/SQS",
                "MetricName", "ApproximateNumberOfMessagesVisible",
                "Dimensions", Collections.singletonList(Map.of(
                        "Name", "QueueName",
                        "Value", Match.anyValue()
                ))
        ));
    }
}

File: typescript/test/dead-letter-queue.test.ts
import { Match, Template } from "@aws-cdk/assertions";
import * as cdk from "@aws-cdk/core";
import { DeadLetterQueue } from "../lib/dead-letter-queue";

describe("DeadLetterQueue", () => {
  test("creates an alarm", () => {
    const stack = new cdk.Stack();
    new DeadLetterQueue(stack, "DeadLetterQueue");

    const template = Template.fromStack(stack);
    template.hasResourceProperties("AWS::CloudWatch::Alarm", {
      Namespace: "AWS/SQS",
      MetricName: "ApproximateNumberOfMessagesVisible",
      Dimensions: [
        {
          Name: "QueueName",
          Value: Match.anyValue(),
        },
      ],
    });
  });
});
